core/store/models/service_agreement.go

```diff
@@ -79,22 +80,28 @@ func NewUnsignedServiceAgreementFromRequest(reader io.Reader) (UnsignedServiceAg
 
 	input, err := ioutil.ReadAll(reader)
 	if err != nil {
-		return UnsignedServiceAgreement{}, err
+		return UnsignedServiceAgreement{}, errors.Wrap(err, "while reading service agreement JSON")
 	}
 
 	err = json.Unmarshal(input, &jsr)
 	if err != nil {
-		return UnsignedServiceAgreement{}, err
+		return UnsignedServiceAgreement{}, errors.Wrap(err, "while parsing job-spec JSON")
 	}
 
 	var encumbrance Encumbrance
 	if err := json.Unmarshal(input, &encumbrance); err != nil {
-		return UnsignedServiceAgreement{}, err
+		return UnsignedServiceAgreement{}, errors.Wrap(err, "while parsing service agreement JSON")
 	}
 
+	if encumbrance.Aggregator == "" {
+		return UnsignedServiceAgreement{}, fmt.Errorf("must set aggregator contract address")
+	}
+	// TODO(alx): Include a check for unset selectors. Need to go back to the JSON
```

Comment:
Please complete this TODO or create a ticket for it.

Reply:
I removed it in fdf92d4969307715cb4d3e287b50f807700a0d1c. Ticket is https://www.pivotaltracker.com/story/show/168439087